src/main/java/com/faforever/client/theme/UiService.java

```diff
@@ -413,6 +413,16 @@ public void loadThemes() {
     return loader.getController();
   }
 
+  public <T extends Controller<?>> T loadFxml(String relativePath, Class<?> controllerClass) {
+    FXMLLoader loader = new FXMLLoader();
+    loader.setControllerFactory(applicationContext::getBean);
```

Comment:
This was only because the fx:include of the searchBox in the entity needed an application bean provided by the factory but removing that will remove the need for the controller factory as well.

Reply:
Just for context to remove the need to set the controller factory would just need to load and place the fxml for the search, logical node and specification. 
What do you think the adverse effect of setting the controller factory vs manually loading and setting the controller for those?

Reply:
ControllerFactory uses the fx:controlle=? in the fxml file and makes a new instance of whatever class there is mentioned. U now just set your own instance that u want.

Reply:
Yes the loader sets the controller for the top level file however the search box has its own set of controllers that also need to be set which are gotten by the controller factory unless we change them to all be handed directly in each of those as well.